src/main/java/de/hsrm/vegetables/my_food_coop_service/services/BalanceHistoryService.java

```diff
@@ -27,12 +28,42 @@
     private final BalanceHistoryItemRepository balanceHistoryItemRepository;
 
     /**
-     * Find balance history items for a user within the specified date range
+     * Find all balance history items for a user within the specified date range
      *
      * @param userDto The user who created the balance history items
      * @param fromDate Start of time window of the balance history item list
      * @param toDate End of time window of the balance history item list
-     * @return A list of balance history items created by the given user
+     * @return A list of all balance history items created by the given user
+     */
+    public List<BalanceHistoryItemDto> findAllByUserDtoAndCreatedOnBetween(
+            UserDto userDto, LocalDate fromDate, LocalDate toDate) {
+
+        LocalDate today = LocalDate.now();
+
+        if (fromDate.isAfter(today) || toDate.isAfter(today)) {
+            throw new BadRequestError("Report Date cannot be in the future", ErrorCode.REPORT_DATA_IN_FUTURE);
+        }
+
+        if (fromDate.isAfter(toDate)) {
+            throw new BadRequestError("fromDate cannot be after toDate", ErrorCode.TO_DATE_AFTER_FROM_DATE);
+        }
+
+        OffsetDateTime fromDateConverted = OffsetDateTime.of(fromDate, LocalTime.MIN, ZoneOffset.UTC);
+        OffsetDateTime toDateConverted = OffsetDateTime.of(toDate, LocalTime.MAX, ZoneOffset.UTC);
```

Comment:
This whole section of code is doubled from the same function that just calls `BalanceHistoryItemRepository::findAllByUserDtoAndCreatedOnBetween`.

Combine these lines into a void function that does nothing if the input data is fine and throws an error if not

Reply:
Done, created a common utility class `Util` with a static method `checkDateRange(fromDate, toDate)` and called it from both places.